torchsde/_brownian/derived.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import brownian_base
+from . import brownian_interval
+from ..types import Optional, Scalar, Tensor
+
+
+class ReverseBrownian(brownian_base.BaseBrownian):
+    def __init__(self, base_brownian):
+        super(ReverseBrownian, self).__init__()
+        self.base_brownian = base_brownian
+
+    def __call__(self, ta, tb=None, return_U=False, return_A=False):
+        # Whether or not to negate the statistics depends on the return value of the adjoint SDE. Currently, the adjoint
+        # returns negated drift and diffusion, so we don't negate here.
+        return self.base_brownian(-tb, -ta, return_U=return_U, return_A=return_A)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(base_brownian={self.base_brownian})"
+
+    @property
+    def dtype(self):
+        return self.base_brownian.dtype
+
+    @property
+    def device(self):
+        return self.base_brownian.device
+
+    @property
+    def shape(self):
+        return self.base_brownian.shape
+
+    @property
+    def levy_area_approximation(self):
+        return self.base_brownian.levy_area_approximation
+
+
+class BrownianPath(brownian_interval.BrownianInterval):
```

Comment:
Is it possible to use some form of dependency injection here, as opposed to inheriting directly from BI? BP should inherit from BaseB, since the latter specifies the API. But inheriting from BI could be a rather strong dependency.

Reply:
Dependency injection, probably not. If you mean composition-over-inheritance, then yes that's possible - I was mostly just avoiding lots of redirections `def blah(self): return self._interval.blah()`, but I can change that.

Reply:
Yes, composition-over-inheritance is what I mean. The motivation is the old-school argument that in the case of inheritance, changing BI's API would affect BPath and BTree.